Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2340,111 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-howto-row-factory:
+
+How to create and use row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represent fetched rows as :class:`tuples <tuple>`.
```

Comment:
```suggestion
By default, :mod:`!sqlite3` represents fetched rows as :class:`tuple`\s.
```

* Fix grammar
* Fix display of class name (as currently it is displayed as the _literal_ `tuples`, which is not the literal class name.

Or, you could do

```suggestion
By default, :mod:`!sqlite3` represents each fetched row as a :class:`tuple`.
```

Reply:
IME using escaped-slash-then-letters is older and clumsier (to write and to read, with change of font stye mid-word) than ```:role:`text <ref>` ```

It is fine to have `tuples` in the code/class markup.

Reply:
Yes, the appearance is (arguably) not ideal, something which I almost made an explicit note of above. However, using custom text that is neither code nor a literal, yet gets rendered as such (`tuples` rather than `tuple`s) is neither semantically correct nor consistent with how code literal formatting is used everywhere else. At least in my view, clarity, consistency and correctness trumps personal aesthetic preference.

Also, `tuple`s is also consistent with the syntax used elsewhere in the `sqlite3` docs, and what I've generally seen in use elsewhere in the Python docs these days for literal cross-reference roles (though that latter is purely anecdotal). Its also more concise to read and write in the source.

Furthermore, it seems a moot point, as I presented an equivalent alternative that avoids this issue entirely.

> older 

Do you mean that support for explicit (custom) titles for Sphinx domain references was not implemented when they were first introduced? I'm not really sure how that would be germane here...

Reply:
I understood the points you made in your first paragraph in your previous message, no need to insist at length!
My message expresses disagreement with these points 🙂  I truly believe there is no problem or confusion created by seeing `tuples` in code or function/class reference markup.  Absolute purity of perfect markup is not a goal without concrete benefits!

P2: internal consistency can be a point, but other instances can also be replaced if different markup is judged better.
For this case, it seems fine to keep the slightly ugly markup with `\ s` then.

P3: yes I have a memory of custom titles not being supported by all roles, or maybe it was that python-dev didn’t know about it and we found `\ s` unsatisfying (clumsy to write and inelegant output).

Reply:
Well, my message was expressing disagreement with your message expressing disagreement with my points  :slightly_smiling_face: , but you're right, its better to just agree to disagree, and leave Erlend to decide. Sorry if I came off as too argumentative, and thanks for being kind about it!

> P2: internal consistency can be a point, but other instances can also be replaced if different markup is judged better.

That's quite true, yeah (though perhaps a little out of scope of this PR).

>  For this case, it seems fine to keep the slightly ugly markup with \ s then.

What do you think about the other suggestion that avoids it?

> were then happy to see the cleaner text that you want in context <ref-target>.

Yeah, the only issue here for me is the `literal` formatting (which is a problem for other custom titles that aren't a literal/code). It might be ideal with these roles offered some per-usage way of triggering not using literal formatting when an explicit title is passed, like `~` and `!` do for other things.

> or maybe it was that python-dev didn’t know about it 

BTW on that note,, I didn't know this for a while myself, but you can do `\s` rather than `\ s`. Doesn't change the output, but at least it saves you a character.

I _believe_ may actually be [an experimental docutils mode](https://github.com/docutils/docutils/blob/bc9bf5fc2c7f905507a49e8e37c12368c2f4a0cc/docutils/docutils/parsers/rst/__init__.py#L138) that actually allows you to do `` :class:`tuple`s ``, though it runs the risk of requiring escaping for special characters more places in plain text (and I'm not sure if its usable nowadays with Sphinx).

Reply:
For the alternate suggestion, I do like `as a tuple` but am not a fan of the circumvolution `each fetched row`. 🙂

Reply:
- I'll got at lengths to avoid the ugly markup (`\ s`)
- IMO using `tuples <tuple>` is probably not a real problem for our average docs reader/consumer
- I think CAMs second suggestion is a very nice solution, so I'll use that and resolve this conversation :)

Reply:
> I'll got at lengths to avoid the ugly markup (\ s)

Just FYI, not that it makes a _huge_ difference, but the sequence in question is just `\s`, not `\ s` (you can escape the `s` directly without an intervening space, which makes it a little less ugly and awkward, at least in the source). Mentioning this again since it seems to be a common misconception (which until relatively recently I had myself).